ListApp/ListApp.Api/App_Start/WebApiConfig.cs

```diff
@@ -11,12 +10,22 @@ public static void Register(HttpConfiguration config)
         {
             // Web API configuration and services
 
+            var constraintResolver = new DefaultInlineConstraintResolver
+            {
+                ConstraintMap =
+                {
+                    ["apiVersion"] = typeof( ApiVersionRouteConstraint )
+                }
+            };
+
+            config.AddApiVersioning();
+
             // Web API routes
-            config.MapHttpAttributeRoutes();
+            config.MapHttpAttributeRoutes(constraintResolver);
 
             config.Routes.MapHttpRoute(
```

Comment:
One does not need default route when using `attribute routing` - it might hide some errors (like missing route attributes) - generally, use either one or the other.